### ISSUE TITLE
create_basesystem_virtual_disk stage issue

### ISSUE THREAD

**Issue by mstaragrieb** [myspaghetti/macos-virtualbox#264]
Line 647 has dmg2img command but I needed to add "./"

**Comment by myspaghetti** [comment]
Thanks for the bug report! If you don't mind, please post the file `macOS_troubleshoot.txt` that is generated by `./macos-guest-virtualbox troubleshoot`

The script runs `dmg2img.exe` from the working directory if it is detected on Cygwin. All other systems should have `dmg2img` installed through the package manager.

**Comment by myspaghetti** [comment]
Thanks for the bug report! If you don't mind, please post the file `macOS_troubleshoot.txt` that is generated by `./macos-guest-virtualbox.sh troubleshoot`

The script runs `dmg2img.exe` from the working directory if it is detected on Cygwin. All other systems should have `dmg2img` installed through the package manager.

**Comment by myspaghetti** [comment]
Thanks for the bug report! If you don't mind, please post the file `macOS_troubleshoot.txt` that is generated by `./macos-guest-virtualbox.sh troubleshoot`

The script runs `dmg2img.exe` from the working directory if the executable is detected on Cygwin. All other systems should have `dmg2img` installed through the package manager.

**Comment by mstaragrieb** [comment]
Well, weird, I could not recreate the bug. My only guess is that I didn't restart after installing cygwin. I will close.

 I was unable to connect to icloud stuff (messages, etc) once I got macOS running, did you ever experience that?

**Comment by myspaghetti** [comment]
Yes, iMessage etc require genuine or genuine-like Apple parameters, see the documentation.

Cygwin doesn't require a restart but it's possible there was some error detecting the script was running on Cygwin.

**Comment by mstaragrieb** [comment]
Oh, okay, that makes sense, I read that but I was so neck deep in it that I didn't put that together, thanks! I am very impressed with this script.

**Comment by mstaragrieb** [comment]
Well shoot, I managed to recreate it and then ran the troubleshoot.
[macOS_troubleshoot.txt](https://github.com/myspaghetti/macos-virtualbox/files/4907905/macOS_troubleshoot.txt)

**Comment by myspaghetti** [comment]
Coule be that the `PWD` variable is not set because the script is not run in a login and interactive shell: `bash --login -i`  
`echo ${PWD}` should return the current working directory; you can add that line to the top of the script to see if it's set correctly.

Somewhat less likely is a permissions issue, `ls -l dmg2img.exe` should list the active user as the owner and the file should be executable:
```
ls -l dmg2img.exe
-rwx------+ 1 Jack None 112128 Feb  6  2017 dmg2img.exe
```

**Comment by RoyiAvital** [comment]
Any chance to set static image instead of dynamic (For better performance)?

**Comment by myspaghetti** [comment]
@RoyiAvital the VM performance bottleneck is the graphics, but if you still want fixed virtual disk images you can edit the script and replace `--variant standard` with `--variant fixed` in lines `708` and `743`. Since the performance improvement is negligible there are no current plans to add this configuration to the script.

Please create a new issue next time, each individual thread should be limited in scope to more or less the topic of the first post.

**Comment by myspaghetti** [comment]
Closing until further information is provided.

**Comment by myspaghetti** [comment]
Duplicate of  #283, fixed in version 0.96.6
